ttnn/cpp/ttnn/operations/conv/conv2d/device/conv2d_op.cpp

```diff
@@ -216,6 +248,36 @@ operation::ProgramWithCallbacks OptimizedConvNew::create_program(const std::vect
         enable_split_reader,
         enable_subblock_padding,
         use_non_tile_height);
+
+        const uint32_t post_op_l1_stats = device->get_memory_allocation_statistics(tt::tt_metal::BufferType::L1).total_allocated_bytes;
+        auto actual_cb_size = program_with_cbs.program.get_cb_memory_size();
+
+        auto [calc_output_size, calc_CB_size] = estimate_L1_usage(
+                arch, this->memory_config.memory_layout,
+                input_dtype, weights_dtype, output_dtype,
+                compute_kernel_config,
+                block_config, parallelization_config,
+                input_tensor_shape, weights_shape, sliding_window_config.get_output_shape(),
+                output_channels, groups, std::array<uint32_t,2>({sliding_window_config.window_hw.first, sliding_window_config.window_hw.second}),
+                Conv2dConfig{
+                    .output_layout = (untilize_out ? Layout::ROW_MAJOR : Layout::TILE),
+                    .enable_act_double_buffer = enable_act_double_buffer,
+                    .enable_weights_double_buffer = enable_weights_double_buffer,
+                    .enable_split_reader = enable_split_reader,
+                    .enable_subblock_padding = enable_subblock_padding
+                }, this->memory_config,
+                has_bias, use_non_tile_height);
+    if(calc_CB_size > 0) {
+        if(calc_CB_size != actual_cb_size) {
```

Comment:
please use proper formatting (space between `if` and `(`, everywhere.

Reply:
Done.